### PR TITLE
Add QuestPatcher v2.2.0

### PR DESCRIPTION
**Pull request by burritosoftware**
- [x] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [x] Have you validated your manifest locally with `winget validate --manifest <path>`? 
- [x] Have you tested your manifest locally with `winget install --manifest <path>`?
- [x] Does your manifest conform to the [1.0 schema](https://github.com/microsoft/winget-cli/blob/master/doc/ManifestSpecv1.0.md)?

Note: `<path>` is the name of the directory containing the manifest you're submitting.

-----

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/winget-pkgs/pull/20163)

### CODE REVIEW

File: manifests/l/Lauriethefish/QuestPatcher/2.2.0/Lauriethefish.QuestPatcher.locale.en-US.yaml [hunk 1]
```diff
@@ -0,0 +1,26 @@
+﻿# Created using YamlCreate.ps1
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.defaultlocale.1.0.0.schema.json
+
+PackageIdentifier: Lauriethefish.QuestPatcher
+PackageVersion: 2.2.0
+PackageLocale: en-US
+Publisher: Lauriethefish
```
Comment (ImJoakim) [inline manifests/l/Lauriethefish/QuestPatcher/2.2.0/Lauriethefish.QuestPatcher.locale.en-US.yaml:7]: image

```suggestion
Publisher: Lauriethefish#6700
```

File: manifests/l/Lauriethefish/QuestPatcher/2.2.0/Lauriethefish.QuestPatcher.installer.yaml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+- InstallerLocale: en-US
+  Architecture: x64
+  InstallerType: inno
+#  Scope: 
+  InstallerUrl: https://github.com/Lauriethefish/QuestPatcher/releases/download/2.2.0/QuestPatcher-windows.exe
+  InstallerSha256: 6B679A74E620C49E05077A7F430EB407336A9E137EEC8E72A4CC58037735E60B
+  InstallerSwitches:
+    Silent: /VERYSILENT /ALLUSERS
+    SilentWithProgress: /SILENT
+  UpgradeBehavior: install
```
Comment (ImJoakim) [inline manifests/l/Lauriethefish/QuestPatcher/2.2.0/Lauriethefish.QuestPatcher.installer.yaml:21]: Let's add user scope for compatibility reasons.

```suggestion
- InstallerLocale: en-US
  Architecture: x64
  InstallerType: inno
  Scope: machine
  InstallerUrl: https://github.com/Lauriethefish/QuestPatcher/releases/download/2.2.0/QuestPatcher-windows.exe
  InstallerSha256: 6B679A74E620C49E05077A7F430EB407336A9E137EEC8E72A4CC58037735E60B
  InstallerSwitches:
    Silent: /VERYSILENT /ALLUSERS
    SilentWithProgress: /SILENT /ALLUSERS
  UpgradeBehavior: install
- InstallerLocale: en-US
  Architecture: x64
  InstallerType: inno
  Scope: user
  InstallerUrl: https://github.com/Lauriethefish/QuestPatcher/releases/download/2.2.0/QuestPatcher-windows.exe
  InstallerSha256: 6B679A74E620C49E05077A7F430EB407336A9E137EEC8E72A4CC58037735E60B
  InstallerSwitches:
    Silent: /VERYSILENT /CURRENTUSER
    SilentWithProgress: /SILENT /CURRENTUSER
```

**Comment by ImJoakim** [comment]
Please take a look at my comments 😃

**Comment by burritosoftware** [comment]
> Please take a look at my comments 😃
> 
> * [Add QuestPatcher v2.2.0 #20163 (comment)](https://github.com/microsoft/winget-pkgs/pull/20163#discussion_r665732521)
> * [Add QuestPatcher v2.2.0 #20163 (comment)](https://github.com/microsoft/winget-pkgs/pull/20163#discussion_r665734948)

Committed both of your suggestions. I was kinda confused on how user scopes worked: thanks for helping with that :D

**Comment by ImJoakim** [comment]
> > Please take a look at my comments 😃
> > 
> > * [Add QuestPatcher v2.2.0 #20163 (comment)](https://github.com/microsoft/winget-pkgs/pull/20163#discussion_r665732521)
> > * [Add QuestPatcher v2.2.0 #20163 (comment)](https://github.com/microsoft/winget-pkgs/pull/20163#discussion_r665734948)
> 
> Committed both of your suggestions. I was kinda confused on how user scopes worked: thanks for helping with that :D

No worries, I'm happy to help :)


Also, some useful information: the reason why the installation failed was because it needed the the `/ALLUSERS` in the SilentWithProgress because of this option when installing:

image

image

**Review by OfficialEsco — APPROVED** [review]
Tips for both:
You only need to add Silent and SilentWithProgress switches if the InstallerType is EXE, /ALLUSERS and /CURRENTUSER can go in the Custom InstallerSwitch 😁

**Comment by burritosoftware** [comment]
> Tips for both:
> You only need to add Silent and SilentWithProgress switches if the InstallerType is EXE, /ALLUSERS and /CURRENTUSER can go in the Custom InstallerSwitch 😁

Will do when an update releases for QP. Thanks! :)

**Pull request merged**